aspnet/mvc/models/file-uploads.rst

```diff
@@ -0,0 +1,197 @@
+File Uploads
+============
+
+By `Steve Smith`_
+
+ASP.NET MVC actions support uploading of one or more files, using simple model binding for smaller files, or streaming for larger files.
+
+.. contents:: Sections
+  :local:
+  :depth: 1
+
+`View or download sample from GitHub <https://github.com/aspnet/Docs/tree/master/aspnet/mvc/models/file-uploads/sample>`_.
+
+Uploading Small Files with Model Binding
+----------------------------------------
+
+To upload small files, you can use a multi-part HTML form or construct a POST request using JavaScript. An example form using Razor which supports multiple uploaded files is shown below:
+
+.. literalinclude:: file-uploads/sample/FileUploadSample/Views/Home/Index.cshtml
+  :language: html
+  :lines: 7-20
+  :dedent: 8
+
+
+In order to support file uploads, HTML forms must specify an ``enctype`` of ``multipart/form-data``. The ``files`` input element shown abovesupports uploading multiple files; omit the ``multiple`` attribute on this input element to allow just a single file to be uploaded. The above markup renders in a browser as:
+
+.. image:: file-uploads/_static/upload-form.png
+
+The individual files uploaded to the server can be accessed through :doc:`model-binding` using the :dn:iface:`~Microsoft.AspNetCore.Http.IFormFile` interface. ``IFormFile`` has the following structure:
+
+.. code-block:: c#
+
+  public interface IFormFile
+  {
+    string ContentType { get; }
+    string ContentDisposition { get; }
+    IHeaderDictionary Headers { get; }
+    long Length { get; }
+    string Name { get; }
+    string FileName { get; }
+    Stream OpenReadStream();
+    void CopyTo(Stream target);
+    Task CopyToAsync(Stream target, CancellationToken cancellationToken = null);
+  }
+
+When uploading files using model binding and the ``IFormFile`` interface, the action method can accept either a single ``IFormFile`` or an ``IEnumerable<IFormFile>`` (or ``List<IFormFile>``) representing several files. The following example loops through one or more uploaded files, saves them to the local file system, and then returns the total number and size of files uploaded.
+
+.. literalinclude:: file-uploads/sample/FileUploadSample/Controllers/UploadFilesController.cs
+  :language: c#
+  :emphasize-lines: 21,27,34
+
+Files uploaded using the ``IFormFile`` technique are buffered in memory on the web server before being processed. Inside the action method, the ``IFormFile`` contents are accessible as a stream. In addition to the local file system, files can be streamed to `Azure blob storage <https://azure.microsoft.com/en-us/documentation/articles/vs-storage-aspnet5-getting-started-blobs/>` or Entity Framework.
+
+To store binary file data in a database using Entity Framework, define a property of type ``byte[]`` on the entity:
+
+.. code-block:: c#
```

Comment:
No, those are only shown as snippets. Didn't want to clutter the sample with the need for EF and a database.